walkeeper/src/receive_wal.rs

```diff
@@ -136,20 +90,33 @@ impl<'pg> ReceiveWalConn<'pg> {
             _ => bail!("unexpected message {:?} instead of greeting", msg),
         }
 
-        // Need to establish replication channel with page server.
-        // Add far as replication in postgres is initiated by receiver, we should use callme mechanism
-        if let Some(ref pageserver_connstr) = self.pageserver_connstr {
-            let conf = swh.conf.clone();
-            let timelineid = swh.timeline.get().timelineid;
-            // copy to safely move to a thread
-            let pageserver_connstr = pageserver_connstr.to_owned();
-            let _ = thread::Builder::new()
-                .name("request_callback thread".into())
-                .spawn(move || {
-                    request_callback(conf, pageserver_connstr, timelineid, tenant_id);
+        // if requested, ask pageserver to fetch wal from us
+        // as long as this wal_stream is alive, callmemaybe thread
+        // will send requests to pageserver
+        let _guard = match self.pageserver_connstr {
+            Some(ref pageserver_connstr) => {
+                // Need to establish replication channel with page server.
+                // Add far as replication in postgres is initiated by receiver, we should use callme mechanism
+                let timelineid = swh.timeline.get().timelineid;
+                let tx_clone = swh.tx.clone();
+                let pageserver_connstr = pageserver_connstr.to_owned();
+                swh.tx
+                    .send(CallmeEvent::Subscribe(
+                        tenant_id,
+                        timelineid,
+                        pageserver_connstr,
+                    ))
+                    .unwrap();
```

Comment:
It can fail when channel was closed or receiver was dropped, thats unlikely but lets add some context via expect or turn this to an error